src/convertToObject.js

```diff
@@ -6,7 +6,44 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString === '') {
+    return {};
+  }
+
+  const modifiedStr = sourceString
+    .trim()
+    .split(';')
+    .map((item) => {
+      return item.trim();
+    });
+
+  if (modifiedStr.length === 1) {
+    const [key, value] = sourceString.trim().split(':');
```

Comment:
The condition `modifiedStr.length === 1` may not correctly handle cases where there is more than one key-value pair in the string. Consider revising this logic to ensure all key-value pairs are processed.